SimDataFormats/CaloAnalysis/interface/SimCluster.h

```diff
@@ -209,11 +223,34 @@ class SimCluster {
   /** @brief clear the energies list */
   void clearHitsEnergy() { std::vector<float>().swap(energies_); }
 
+  /** @brief clear the times list */
+  void clearHitsTime() { std::vector<float>().swap(times_); }
+
+  void clear() {
+    clearHitsAndFractions();
+    clearHitsEnergy();
+    clearHitsTime();
+  }
+
+  /** @brief computes the time of the cluster */
+  float computeClusterTime() const {
```

Comment:
Are we sure such a time assignment makes sense also for calorimeters, like HGCAL? If not and if it's specific to MTD, maybe it would be better to move it to some derived class?

Reply:
@rovere this calculation is that used in the MTD clusterization, nothing particularly fancy (a energy-weighted time average), but HGCAL experts should comment whether this makes sense of that detector

Reply:
would a new class like `MtdSimCluster` that inherits from the `SimCluster` and adds the time related methods be ok for you?

Reply:
I believe it is up to main users to reply here, IMHO it might make sense. Alternatively the method could be made subdetector dependent, but I believe you miss this piece of information inside this object, which might be unpractical if you want to extend it beyond the only HGCAL. I let other to comment about this.

Reply:
@AuroraPerego could you please clarify the strategy? I see that now ```MtdSimLayerCluster``` does not inherit any more from ```SimCluster```, but the time definition is still here. Is it clear if this is ok for all the stakeholders?